tests/unit/hosts/test_views.py

```diff
@@ -573,3 +573,135 @@ def test_when_requesting_user_is_not_part_of_team(self):
         response = self.client.post(self.url, self.data)
         self.assertEqual(response.data, expected)
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+
+class DeleteChallengeHostFromTeamTest(BaseAPITestClass):
```

Comment:
How is this different from `DeleteParticularChallengeHost`? (https://github.com/Cloud-CV/EvalAI/blob/master/tests/unit/hosts/test_views.py#L429)

Reply:
> How is this different from `DeleteParticularChallengeHost`? (https://github.com/Cloud-CV/EvalAI/blob/master/tests/unit/hosts/test_views.py#L429)

@nikochiko in DeleteParticularChallengeHost, there was no check if admin decided to delete himself and so it will delete the whole host team if admin tried to delete himself. So i wrote another test which checks if it's the admin who tries to delete himself/herself and if so then show him/her an error message and deny the request by showing appropriated message. 

Also i was not sure why DeleteParticularChallengeHost was written so instead of editing it i preferred writing a new one.

Reply:
So as in [this](https://github.com/Cloud-CV/EvalAI/pull/2660/files#r377025730) conversation, it is probably a bug. 
But having two views for performing the same action would be confusing - see [Single Responsibility Principle](https://codeburst.io/the-single-responsibility-principle-2c09c279fce9?gi=3d98d7da3669). 
And same thing with two classes for testing the same functionality, what do you think about merging these two classes?

Reply:
> So as in [this](https://github.com/Cloud-CV/EvalAI/pull/2660/files#r377025730) conversation, it is probably a bug.
> But having two views for performing the same action would be confusing - see [Single Responsibility Principle](https://codeburst.io/the-single-responsibility-principle-2c09c279fce9?gi=3d98d7da3669).
> And same thing with two classes for testing the same functionality, what do you think about merging these two classes?

I was also thinking the same but was not sure. I will just open up an issue for previously written view and url and will edit there only to make the deletion functionality work.